File: Cpp_src/easy/hamming_distance_461.cc
#include <iostream>
using namespace std;

// Use bit manipulation
// Time Complexity: O(logn)
// Space Complexity:O(1)
// Runtime: 0 ms, faster than 100.00%
// Memory Usage: 8.1 MB, less than 95.45%
class Solution
{
public:
    int hammingDistance(int x, int y)
    {
        int tmp = x ^ y;
        int result = 0;
        while (tmp != 0)
        {
            tmp &= tmp - 1;
            result++;
        }
        return result;
    }
};

File: Golang_src/easy/hammingDistance_461.go
package easy

// Use bit manipulation:
// Time Complexity: O(logn)
// Space Complexity:O(1)
// Runtime: 0 ms, faster than 100.00%
// Memory Usage: 2 MB, less than 100.00%
func hammingDistance(x int, y int) int {
	tmp := x ^ y
	result := 0

	for tmp != 0 {
		result++
		tmp &= tmp - 1
	}

	return result
}
